pkg/apis/build/v1alpha1/build_validation.go

```diff
@@ -92,3 +96,38 @@ func (bt *BuildSpec) validateTimeout() *apis.FieldError {
 	}
 	return nil
 }
+
+// Validate source
+func (bs BuildSpec) validateSources() *apis.FieldError {
+	var subPathExists bool
+	names := map[string]string{}
+	var sourceExists bool
+
+	if bs.Source != nil {
+		sourceExists = true
+	}
+	if bs.Sources != nil && sourceExists {
+		return apis.ErrMultipleOneOf("b.spec.source", "b.spec.sources")
+	}
+	for _, source := range bs.Sources {
+		// check all source have names
+		if source != nil {
+			if source.Name == "" {
```

Comment:
Does it make sense to allow checking out one source (probably the first) to /workspace root (not sure the answer)?

Reply:
I did not consider this use case. I guess you want to use `sources` but want the same behavior as `source` where it is present in `/workspace` ? (Just to make sure I understood your question correctly)

Great question. I do not have strong opinion about continuing to use `source` or support this feature in `sources`. Do you have an opinion about this

Reply:
I don't have an opinion about this, but thought I'd mention it because builds seem to generally assume that PWD has the primary source in it. So I suspect it will be fairly common to have your primary source (aka the repo with the PR that needs to be tested) plus any additional repos the test needs to include. 

If I start out with a single repo and then need to add a second repo later, currently I'll need to update all the steps to set `WorkDir: /workspace/first-repo-name`.

The way our cloning container handles this is by special casing the "primary" ref and defaulting the working directory to this /workspace/go/src/org/repo subpath rather than /workspace 
 (see https://github.com/kubernetes/test-infra/pull/9907)

So it seems like it would be nice if we either allow checking out a repo to /workspace or else specifying which source I want to set my workdir to.

Reply:
If we use `TargetPath` field and if that field is not set then source will be present in `/workspace`. 
One problem I see with this approach is build with 2 sources and no targetPaths will dump contents in `/workspace` and if there is any clash of file names, user will not realize this I suppose.

Reply:
Nevermind. This will be be an error